### ISSUE TITLE
⌨ Redundant link announcement in resource block

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#14300]
### ⌨ Accessibility and Expected Behaviour

Our resource block has the list item role along with an aria label. The issue is, the aria label is mostly redundant causing the link test to be read out in duplicate.

### 🕵️ Details

The aria label is important becuase it adds information to the item (current state) however, since it is on a list item, it gets read out before the link and you hear the link text twice.

Eg. Link, Personal information, Personal information (complete), group

### What operating systems are you encountering the problem on?

macOS

### What browsers are you encountering the problem on?

Safari

### What assistive technologies are you encountering the problem on?

VoiceOver

### 📸 Screenshot

<img width="1510" height="774" alt="Image" src="https://github.com/user-attachments/assets/7b1abc9d-ec1a-4a89-9d84-9dc1faf7c3dc" />

### 🙋‍♀️ Proposed Solution

I think it should be safe to move the `aria-label` to the link instead of the list item.

### ✅ Acceptance Criteria

- [ ] Link text is not duplicated
